src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java

```diff
@@ -128,26 +129,50 @@ boolean amendIfNecessary(
       return false;
     }
 
-    for (SinkRecordField missingField: missingFields) {
-      if (!missingField.isOptional() && missingField.defaultValue() == null) {
+    // At this point there are missing fields
+    TableType type = tableDefn.type();
+    switch (type) {
+      case TABLE:
+        // Rather than embed the logic and change lots of lines, just break out
+        break;
+      case VIEW:
+      default:
         throw new ConnectException(
-            "Cannot ALTER to add missing field " + missingField
-            + ", as it is not optional and does not have a default value"
+            String.format(
+                "%s %s is missing fields (%s) and ALTER %s is unsupported",
+                type.capitalized(),
+                tableId,
+                missingFields,
+                type.jdbcName()
+            )
```

Comment:
nice check! since the view schema cannot be altered by the Connector, should we not ignore `auto.evolve` if the table to be written to is a view, and simply drop all the new fields? or have a validation check that fails config validation if we try to set auto.enable when table.types includes views (which may be too strict actually)? either way, I feel it doesn't make sense to bundle view and table updates in a single deployed connector. users will have to spin up different connectors: one for views that can manage schema evolution, and the other one where it is just not supported.

WDYT?

Reply:
When records for this "table" are being sent to a view, the proposal is to require that the view exactly match the record value fields, per the check above. IOW, in that case we *are* ignoring the `auto.evolve` setting.

Maybe I don't understand what you're asking here.

Reply:
Okay, so I talked with @wicknicks offline, and came to these conclusions:

1. If the record has extra fields relative to the table, the connector will use `auto.evolve` to determine whether to attempt to alter the table to match the record or fail with an exception stating that fields are missing from the table. 
2. If the record has extra fields relative to the view, the connector will fail with an exception stating that fields are missing from the view. This reflects the table behavior (with `auto.evolve=false`), since for views `auto.evolve` is ignored as the connector never has enough information to alter the view definition.
3. If the record is missing some columns from the table or view, the connector does not catch this and instead the operations will succeed iff the extra columns have defaults and/or are nullable, but will fail with a runtime exception if the extra columns are required. This is kind of an undocumented feature, and it would actually be somewhat difficult to explain this corner case in the config docs.

Reply:
Thanks for the summary, @rhauch. For points 1 and 3, it makes sense. For 2, though, I don't believe we ignore `auto.evolve` when the destination is a sink. The task crashes with a ConnectException (as seen even in the `PostgresViewIT#testRecordSchemaMoreFieldsThanViewFails` test). 

It's ok to do this, but we just need to set the right expectation. Effectively, users of the connector might have to run separate connectors to sink records to both views and tables. That is only way to enable auto evolution for tables, and disable for the views in the pipeline.

Reply:
> Effectively, users of the connector might have to run separate connectors to sink records to both views and tables.

Remember that `auto.evolve` only applies to tables, and is important only when the columns of *each* table do not match the record. The behavior of writing to views is independent of this setting.

But it is possible to run one connector that writes to views and tables with `auto.evolve=true` and to have the sink connector ALTER tables as needed.

Consider an example where a user runs one JDBC sink connector that sinks to both a table and a view, and they set:
```
auto.evolve=true
table.types=TABLE,VIEW
```
If the record has, say, one more field than what corresponds to the table's columns, the connector will (per `auto.evolve=true`) attempt to ALTER the table to add the missing column. If the database user has privilege to do this, then this should work and the connector should be able to then write the record to the table.

Meanwhile, if the same connector consumed a record that corresponded to the view and all record fields existed in the view, then the connector would be able to write to the view.

I will attempt to address this in a forthcoming DOCS PR.

Reply:
In your example, what happens if the records that go to the view added new fields. that would trigger an ALTER query, and a subsequent task failure, right?

Reply:
If the sink consumed a record destined for the view, and the view were missing some columns in the record, the sink connector will fail with an error message similar to the following, regardless of the `auto.evolve` setting:
```
VIEW `customers` is missing fields (SinkRecordField{schema=INT64, name='essential_column', isPrimaryKey=false}) and ALTER VIEW is unsupported
```

Reply:
makes sense. and that's the bit you wanted to add to documentation?

Reply:
Yes. It's impossible to have that much detail in the config docs, and IMO the code is pretty clear considering the order of the checks in this section.

Reply:
alright cool, as long as the docs are updated to give a hint about this behavior, I think we should be fine. thanks for your time and responses :).

Reply:
Thanks!